tests/v2/validation/provisioning/rke2/custom_cluster_test.go

```diff
@@ -144,13 +145,19 @@ func (c *CustomClusterProvisioningTestSuite) ProvisioningRKE2CustomCluster(exter
 	}
 }
 
-func (c *CustomClusterProvisioningTestSuite) ProvisioningRKE2CustomClusterDynamicInput(externalNodeProvider ExternalNodeProvider, nodesAndRoles []map[string]bool) {
+func (c *CustomClusterProvisioningTestSuite) ProvisioningRKE2CustomClusterDynamicInput(externalNodeProvider ExternalNodeProvider, nodesAndRoles []machinepools.NodeRoles) {
 	rolesPerNode := []string{}
 
 	for _, nodes := range nodesAndRoles {
 		var finalRoleCommand string
-		for role := range nodes {
-			finalRoleCommand += fmt.Sprintf(" --%s", role)
+		if nodes.ControlPlane == true {
```

Comment:
You don't have to do the '==' you can just do `if nodes.ControlPlane` less verbose that way